### ISSUE TITLE
RMSNorm's weight not registered as submodules when initializing

### ISSUE THREAD

**Issue by Darkbblue** [huggingface/diffusers#11938]
### Describe the bug

RMSNorm() look like this, which is in models/normalization.py
```python
class RMSNorm(nn.Module):
    def __init__(self, dim, eps: float, elementwise_affine: bool = True, bias: bool = False):
        super().__init__()

        self.eps = eps
        self.elementwise_affine = elementwise_affine

        if isinstance(dim, numbers.Integral):
            dim = (dim,)

        self.dim = torch.Size(dim)

        self.weight = None
        self.bias = None

        if elementwise_affine:
            self.weight = nn.Parameter(torch.ones(dim))
            if bias:
                self.bias = nn.Parameter(torch.zeros(dim))
        print(self.weight)
        print(self)
```
This module is used in Flux.1-dev, and maybe other models I don't know.
When I print the model, it looks like `(norm_q): RMSNorm()`, which means the self.weight is not registered as submodules.
I've checked that self.weight does exist. Actually, when initialized from Attention, elementwise_affine and bias are never modified, so they are using the default values.
The output looks like this:
```
tensor(..., device='meta', size=(128,), requires_grad=True)
RMSNorm()
```
LLM told me it's because the self.weight is assigned in a conditional branch, AND it's initialized on meta device with from_pretrained(). But it's not really the cause. I removed the branch but the bug persisted.
This bug might not be fatal in most cases, but it prevents me from using FSDP with Flux.

### Reproduction

```python
import torch
from diffusers import FluxPipeline

pipe = FluxPipeline.from_pretrained(
    'black-forest-labs/FLUX.1-dev', torch_dtype=torch.bfloat16
).to('cuda')

prompt = "a cat holding a paper with word prompt on it"
image = pipe(
    prompt,
    height=1024,
    width=1024,
).images[0]
image.save("flux.png")
'''

### Logs

```shell

```

### System Info

diffusers["torch"]==0.32.2

**Comment by DN6** [comment]
@Darkbblue Not sure I follow here? When loading the model with the snippet you shared, the RMSNorm weights are the tensors present in the checkpoint. You wouldn't be able to set the device to cuda if they were on meta device. 

```python
ipdb> pipe.transformer.single_transformer_blocks[0].attn.norm_q
RMSNorm((128,), eps=1e-06, elementwise_affine=True)
ipdb> pipe.transformer.single_transformer_blocks[0].attn.norm_q.weight
Parameter containing:
tensor([1.2266, 1.2578, 1.2969, 1.2734, 1.2500, 1.0547, 1.2812, 1.2891, 1.2422,
        1.2656, 1.2656, 1.2734, 1.2031, 1.2578, 1.1016, 1.2734, 1.3125, 1.4141,
        1.2266, 1.2188, 1.1953, 1.2656, 1.1953, 1.2656, 1.1875, 1.1641, 1.1641,
        1.1172, 1.1797, 1.1406, 1.1641, 1.2266, 1.1797, 1.1484, 1.1953, 1.2031,
        1.2109, 1.1953, 1.2031, 1.2109, 1.2656, 1.2812, 1.2266, 1.1875, 1.2422,
        1.2188, 1.2734, 1.2500, 1.2734, 1.2344, 1.2891, 1.2188, 1.2891, 1.2734,
        1.2734, 1.2734, 1.2812, 1.1172, 1.1094, 1.2188, 1.2578, 1.2891, 1.1875,
        1.0547, 1.1641, 1.2812, 1.1250, 1.2891, 1.2812, 1.2500, 1.2188, 1.2266,
        1.3203, 1.1172, 1.3047, 1.0781, 1.1484, 1.2500, 1.1172, 1.1953, 1.2109,
        1.2109, 1.2500, 1.1484, 1.1406, 1.1875, 1.1172, 1.1797, 1.1641, 1.2344,
        1.2109, 1.2031, 1.2109, 1.1641, 1.2344, 1.2188, 1.2500, 1.2500, 1.2656,
        1.2344, 1.2109, 1.2500, 1.2109, 1.2422, 1.2578, 1.2422, 1.1484, 1.2734,
        1.2656, 1.2578, 1.2891, 1.1406, 1.2734, 1.1250, 1.2891, 1.2734, 1.1172,
        1.2188, 1.2109, 1.2734, 1.2734, 1.2656, 1.2969, 1.2812, 1.2734, 1.2344,
        1.2266, 1.2891], device='cuda:0', dtype=torch.bfloat16,
       requires_grad=True)
ipdb> pipe.transformer.transformer_blocks[0].attn.norm_q.weight
Parameter containing:
tensor([ 2.1851e-02,  6.9141e-01,  3.1836e-01,  1.3672e-02,  4.3945e-01,
         5.6641e-01,  1.7480e-01,  1.0449e-01,  1.0620e-02,  1.9629e-01,
         3.1055e-01,  1.2988e-01,  4.5117e-01,  6.4941e-02,  2.1094e-01,
         3.8086e-01,  1.7871e-01,  2.2656e-01,  1.9434e-01,  3.4570e-01,
         3.3008e-01,  4.0430e-01,  3.1641e-01,  4.4141e-01,  4.9023e-01,
         3.4766e-01,  3.3789e-01,  1.3477e-01,  1.9824e-01,  1.4160e-01,
        -5.2979e-02,  2.3535e-01,  6.8848e-02, -4.3457e-02,  8.5938e-02,
         1.5332e-01,  2.4609e-01,  5.0000e-01,  1.6211e-01,  7.0496e-03,
        -3.9864e-04,  4.8242e-01,  2.0996e-01,  2.4121e-01,  6.9336e-02,
         4.2188e-01,  2.4219e-01,  1.6846e-02,  2.3145e-01,  5.5078e-01,
         4.8438e-01,  4.0820e-01,  3.4570e-01,  2.3730e-01,  1.3379e-01,
         4.7656e-01,  3.0078e-01,  4.6387e-02,  1.5625e-01,  2.5391e-01,
         3.7695e-01,  1.9922e-01,  1.6797e-01,  1.0352e-01,  4.9414e-01,
        -8.9645e-04,  1.6016e-01,  3.8086e-01,  5.9766e-01,  2.7344e-01,
         3.9258e-01,  2.7734e-01,  2.3499e-03,  3.9551e-02,  1.6406e-01,
         1.2402e-01,  6.3965e-02,  3.9648e-01,  1.8262e-01,  3.4766e-01,
         3.4961e-01,  4.0234e-01,  2.2461e-01,  1.6895e-01,  9.7168e-02,
         4.9561e-02,  1.3867e-01,  1.5332e-01,  1.3672e-01,  1.0400e-01,
         9.0332e-02,  8.2031e-02,  2.2559e-01,  8.0566e-02,  3.0212e-03,
         1.8652e-01,  2.5195e-01,  4.9805e-01,  8.0566e-02,  4.9072e-02,
         6.2891e-01,  8.7891e-02,  4.5898e-01,  3.8672e-01,  9.1797e-02,
         3.5742e-01,  6.8359e-02,  5.2344e-01,  3.3203e-01,  6.8970e-03,
        -3.4714e-04,  1.0469e+00,  3.5858e-03,  5.7422e-01,  4.5312e-01,
         8.7500e-01,  4.2773e-01,  4.6484e-01,  2.5391e-01,  4.9023e-01,
         1.5820e-01,  2.7148e-01,  1.3281e-01,  3.2031e-01,  2.3340e-01,
         4.7852e-01,  5.3516e-01,  4.6094e-01], device='cuda:0',
       dtype=torch.bfloat16, requires_grad=True)
```

**Comment by ZhaoChuyang** [comment]
same problem
